common/common-client/src/main/java/org/ow2/proactive/authentication/keycloak/KeycloakOidcProperties.java

```diff
@@ -0,0 +1,201 @@
+ * You must use provided methods in order to get these properties.
+ */
+@PublicAPI
+public class KeycloakOidcProperties {
+
+    /* ***************************************************************** */
+    /* ************************** Keycloak PROPERTIES ****************** */
+    /* ***************************************************************** */
+
+    /** URL of a Keycloak server configured for ProActive authentication */
+    public static final String KEYCLOAK_URL = "keycloak.url";
+
+    /** When Keycloak URL uses the HTTPS protocol, this parameter indicates which kind of TLS certificate to consider for the Keycloak server.
+     * True means that any certificate can be used including the self-signed ones.
+     * False means that only valid certificates generated by Trusted Certification Authorities are allowed.
+     * */
+    public static final String ALLOW_ANY_CERTIFICATE = "keycloak.https.allow_any_certificate";
+
+    /** Keycloak realm in which ProActive client is configured */
+    public static final String KEYCLOAK_REALM = "keycloak.realm";
+
+    /** Keycloak ProActive client ID */
+    public static final String KEYCLOAK_PROACTIVE_CLIENT_ID = "keycloak.proactive.client_id";
+
+    /** Keycloak ProActive client ID */
+    public static final String KEYCLOAK_PROACTIVE_CLIENT_SECRET = "keycloak.proactive.client_secret";
+
+    /** Scope of the information to be returned by Keycloak for the ProActive client.
+     * An empty value considers the scopes defined for ProActive client in Keycloak.
+     */
+    public static final String KEYCLOAK_PROACTIVE_SCOPE = "keycloak.proactive.scope";
+
+    /**  Keycloak RSA public key used to decrypt the access tokens */
+    public static final String KEYCLOAK_PUBLIC_KEY = "keycloak.public.key";
+
+    /**  The attribute in Keycloak access token to be used to extract the user/principal name */
+    public static final String KEYCLOAK_PRINCIPAL_ATTRIBUTE = "keycloak.principal_attribute";
+
+    /**  Use the resource_access roles in Keycloak access token to extract the user/principal roles
+     * true or false */
+    public static final String KEYCLOAK_USE_RESOURCE_ROLE_MAPPINGS = "keycloak.use_resource_role_mappings";
+
+    /** Fall back property, check user/password and group in files if user is not found in Keycloak.
+     * true or false */
+    public static final String FALLBACK_USER_AUTH = "pa.keycloak.authentication.fallback";
+
+    /** group fall back property, check user group membership group file if user is not found in corresponding Keycloak group.
+     * true or false */
+    public static final String FALLBACK_GROUP_MEMBERSHIP = "pa.keycloak.group.membership.fallback";
+
+    /** tenant fall back property, check user tenant membership tenant file if user is not found in LDAP or if tenant Keycloak attribute is not defined.
+     * true or false */
+    public static final String FALLBACK_TENANT_MEMBERSHIP = "pa.keycloak.tenant.membership.fallback";
```

Comment:
As there is no property that defines a tenant attribute in the keycloack configuration, I don't think this is useful.

We might consider though what would be multi-tenant when using keycloak

Reply:
So should I remove it or keep it for a later use (_when we consider multi-tenant based on keycloak_) ?